### ISSUE TITLE
request: update generated code to return nil instead of a nil value

### ISSUE THREAD

**Issue by marcelloh** [a-h/templ#991]
**Describe the bug**
It's not really a bug, but more of a "thing to do, when coding in Go"

In the generated code, there is a lot of:
```
		if templ_7745c5c3_Err != nil {
			return templ_7745c5c3_Err
		}
		return templ_7745c5c3_Err
```

but it is more idiomatic to return just nil, like this:
```
		if templ_7745c5c3_Err != nil {
			return templ_7745c5c3_Err
		}
		return nil
```

**Comment by a-h** [comment]
I'd take a PR for that.

**Comment by a-h** [comment]
Closed in #1025 - thanks for the contribution!
